services/ui-src/src/components/reports/ModalOverlayReportPage.tsx

```diff
@@ -116,93 +87,36 @@ export const ModalOverlayReportPage = ({
   } = useDisclosure();
 
   const openDeleteEntityModal = (entity: EntityShape) => {
-    setCurrentEntity(entity);
+    setSelectedEntity(entity);
     deleteEntityModalOnOpenHandler();
   };
 
   const closeDeleteEntityModal = () => {
-    setCurrentEntity(undefined);
+    clearSelectedEntity();
     deleteEntityModalOnCloseHandler();
   };
 
   // Open/Close overlay action methods
   const openEntityDetailsOverlay = (entity: EntityShape) => {
-    setCurrentEntity(entity);
+    setSelectedEntity(entity);
     setIsEntityDetailsOpen(true);
     setSidebarHidden(true);
   };
 
   const closeEntityDetailsOverlay = () => {
-    setCurrentEntity(undefined);
+    clearSelectedEntity();
     setIsEntityDetailsOpen(false);
     setSidebarHidden(false);
   };
 
-  // Form submit methods
-  const onSubmit = async (enteredData: AnyObject) => {
```

Comment:
This onsubmit was never actually being used anymore. It was orginially there from when we ported over code from MLR, but has since refactored all of this and pushed it to the EntityProvider